tests/experimental/datasources/integration/test_integration_datasource.py

```diff
@@ -111,3 +111,125 @@ def test_sql_query_data_asset(empty_data_context):
+            "yellow_tripdata_sample_2020",
+            "add_splitter_year",
+            {"column_name": "pickup_datetime"},
+            ["year"],
+            1,
+            {"year": 2020},
+            1,
+            {"year": 2020},
+            id="year",
+        ),
+        pytest.param(
+            "yellow_tripdata_sample_2020_all_months_combined.db",
+            "yellow_tripdata_sample_2020",
+            "add_splitter_year_and_month",
+            {"column_name": "pickup_datetime"},
+            ["year", "month"],
+            12,
+            {"year": 2020, "month": 6},
+            1,
+            {"year": 2020, "month": 6},
+            id="year_and_month",
+        ),
+        pytest.param(
+            "yellow_tripdata.db",
+            "yellow_tripdata_sample_2019_02",
+            "add_splitter_year_and_month_and_day",
+            {"column_name": "pickup_datetime"},
+            ["year", "month", "day"],
+            28,
+            {"year": 2019, "month": 2, "day": 10},
+            1,
+            {"year": 2019, "month": 2, "day": 10},
+            id="year_and_month_and_day",
+        ),
+        pytest.param(
+            "yellow_tripdata.db",
+            "yellow_tripdata_sample_2019_02",
+            "add_splitter_datetime_part",
+            {
+                "column_name": "pickup_datetime",
+                "datetime_parts": ["year", "month", "day"],
+            },
+            ["year", "month", "day"],
+            28,
+            {"year": 2019, "month": 2},
+            28,
+            {"year": 2019, "month": 2, "day": 28},
+            id="datetime_part",
+        ),
+        pytest.param(
+            "yellow_tripdata.db",
+            "yellow_tripdata_sample_2019_02",
+            "add_splitter_column_value",
+            {"column_name": "passenger_count"},
+            ["passenger_count"],
+            7,
+            {"passenger_count": 3},
+            1,
+            {"passenger_count": 3},
+            id="column_value",
+        ),
+        pytest.param(
+            "yellow_tripdata.db",
+            "yellow_tripdata_sample_2019_02",
+            "add_splitter_divided_integer",
+            {"column_name": "passenger_count", "divisor": 3},
+            ["quotient"],
+            3,
+            {"quotient": 2},
+            1,
+            {"quotient": 2},
+            id="divisor",
+        ),
+    ],
+)
+def test_column_splitter(
+    empty_data_context,
+    database,
+    table_name,
+    splitter_name,
+    splitter_kwargs,
+    sorter_args,
+    all_batches_cnt,
+    specified_batch_request,
+    specified_batch_cnt,
+    last_specified_batch_metadata,
+):
+    context = empty_data_context
+    datasource = sqlite_datasource(context, database)
+    asset = datasource.add_table_asset(
+        name="table_asset",
+        table_name=table_name,
+    )
+    getattr(asset, splitter_name)(**splitter_kwargs)
+    asset.add_sorters(sorter_args)
```

Comment:
@billdirks It seems that the assertions below would work even without adding the sorters; is this correct?  If not, where should I look in order to see what is actually happening?  Thanks.

Reply:
The assertion at line 235 will fail. I am grabbing the last batch to make assertions. To guarantee which batch is last, I sort them.

Reply:
🙇